pkg/sql/gcjob/table_garbage_collection.go

```diff
@@ -185,3 +186,77 @@ func clearSpanData(
 
 	return nil
 }
+
+// DeleteAllTableData issues a range tombstone over the specified table's data.
+func DeleteAllTableData(
+	ctx context.Context,
+	db *kv.DB,
+	distSender *kvcoord.DistSender,
+	codec keys.SQLCodec,
+	table catalog.TableDescriptor,
+) error {
+	log.Infof(ctx, "deleting data for table %d", table.GetID())
+	tableKey := roachpb.RKey(codec.TablePrefix(uint32(table.GetID())))
+	tableSpan := roachpb.RSpan{Key: tableKey, EndKey: tableKey.PrefixEnd()}
+	return deleteAllSpanData(ctx, db, distSender, tableSpan)
+}
+
+// deleteAllSpanData issues a range tombstones over all data in the specified span.
+func deleteAllSpanData(
+	ctx context.Context, db *kv.DB, distSender *kvcoord.DistSender, span roachpb.RSpan,
+) error {
+
+	// When the distSender receives a DeleteRange request, it determines how the
+	// request maps to ranges, and automatically parallelizes the sending of
+	// sub-batches to the different ranges. If the DeleteRange request spans too
+	// many ranges, the distSender may reach its concurrency limit, causing severe
+	// latency to foreground traffic (see #85470).
```

Comment:
Good catch, thanks! Unfortunate that the DistSender doesn't handle this internally.